backend/app/utils/store_vectors.py

```
@@ -0,0 +1,32 @@
+    """
+    if not vectors:
+        raise ValueError("Vectors list cannot be empty")
+
+    try:
+        index.upsert(vectors, namespace=namespace)
+        logger.info(f"Successfully stored {len(vectors)} "
+                    f"vectors in namespace '{namespace}'")
+    except Exception as e:
+        logger.error("Failed to store "
+                     f"vectors in namespace '{namespace}': {e}")
+        raise RuntimeError(f"Vector storage failed: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)